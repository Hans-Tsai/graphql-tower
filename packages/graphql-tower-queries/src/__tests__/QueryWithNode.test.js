import faker from 'faker';
import { graphql, GraphQLInt, GraphQLSchema, GraphQLObjectType } from 'graphql';
import { QueryWithNode } from '../';

describe('QueryWithNode', () => {
  it('snapshot', async () => {
    const query = new QueryWithNode();
    expect(query).toMatchSnapshot();
  });

  it('successfully query', async () => {
    const resolve = jest.fn((payload, { id }) => id);
    const QueryNode = class extends QueryWithNode {
      type = GraphQLInt;
      resolve = resolve;
    };

    const fieldId = faker.random.number();
    const tempId = faker.random.number();

    const schema = new GraphQLSchema({
      query: new GraphQLObjectType({
        name: 'Query',
        fields: {
          node: {
            type: new GraphQLObjectType({
              name: 'Node',
              fields: {
                field: new QueryNode(),
                custom: new QueryNode('tempId'),
                empty: new QueryNode(),
              },
            }),
            resolve: () => ({ fieldId, tempId }),
          },
        },
      }),
    });

    const result = await graphql(schema, 'query { node { field custom empty } }');
    expect(result.data.node).toEqual({ field: fieldId, custom: tempId, empty: null });
  });
});
