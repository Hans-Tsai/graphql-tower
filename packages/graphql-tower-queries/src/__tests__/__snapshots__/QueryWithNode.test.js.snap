// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryWithNode snapshot 1`] = `
QueryWithNode {
  "_": Object {
    "afterware": Array [],
    "middleware": Array [],
    "name": undefined,
    "parentArgs": Object {
      "id": PayloadField {
        "resolveValue": [Function],
      },
    },
    "resolve": [Function],
    "subscribe": [Function],
  },
  "args": Object {},
  "fieldName": undefined,
  "resolve": [Function],
}
`;
